HW14/HW14.py

```diff
@@ -0,0 +1,60 @@
+import re
+import json
+
+# 1
+with open('phonebook.txt', 'x+', encoding='utf-8') as phone1:
+    try:
+        phone_book = json.load(phone1)
+    except Exception:
+        phone_book = {}
+
+while True:
+    commands = input('Write command:\n')
+    if commands == 'stats':
+        print(f'Amount of records is {len(phone_book)}')
+    elif commands == 'list':
+        print(*phone_book.keys(), sep='\n')
+    elif commands in ['delete', 'show']:
+        name = input('Write name:\n')
+        if commands == 'delete' and name in phone_book.keys():
+            del phone_book[name]
+            with open('phonebook.txt', 'w') as ph:
+                json.dump(phone_book, ph)
+                print(f'Record with name "{name}" was successfully deleted')
+        elif commands == 'show' and name in phone_book.keys():
+            print(f'Here you could see the info for "{name}": {phone_book[name]}')
+        else:
+            print(f'There is no such name as {name} in phone book. Firstly, you need to add it!')
+    elif commands == 'add':
+        new_name, phone = input('Provide, please, name and phone:\n').split()
+        if new_name in phone_book.keys():
+            print(f'Record with {new_name} already exists. Please remove it and add a new one after that')
+        else:
+            if re.match(r'\A(\+380|380|0)[0-9]{7}', phone):
```

Comment:
Your pattern validates valid formats in the wrong way, instead of validating the correct format it validates only one of three options:
```
+380951234567 -> ['0951234567']
0951234567 -> ['0951234567']
380951234567 -> ['0951234567']
```

For invalid formats, there are also some cases when you allow some wrong formats:
```
a0951234567 -> ['0951234567']
8+30951234567 -> ['0951234567']
a+380951234567 -> ['0951234567']
80951234567 -> ['0951234567']
+830965254141 -> ['0965254141']